src/main/java/org/apache/fineract/api/OperationsDetailedApi.java

```diff
@@ -227,74 +227,98 @@ public Page<TransactionRequest> transactionRequests(
-                break;
-            case ERRORDESCRIPTION:
-                data = transactionRequestRepository.filterByErrorDescription(parseErrorDescription(ids));
-                break;
-        }
-        if(filter == Filter.ERRORDESCRIPTION && data != null) {
+            logger.warn("failed to parse dates {} / {}", startFrom, startTo);
+        }
+
+        Specifications<TransactionRequest> spec = null;
+
+        List<TransactionRequest> data = new ArrayList<>();
+        for (String filterBy : filterByList) {
+            List<String> ids = body.get(filterBy);
+            if (ids.isEmpty()) { continue; }
+            Filter filter;
+            try {
+                filter = parseFilter(filterBy);
+                logger.info("Filter parsed successfully " + filter.name());
+            } catch (Exception e) {
+                Map<String, String> res = new HashMap<>();
+                res.put("errorCode", ErrorCode.INVALID_FILTER.name());
+                res.put("errorDescription", "Invalid filter value " + filterBy);
+                res.put("developerMessage", "Possible filter values are " + EnumSet.allOf(Filter.class));
+                logger.info("Unable to parse filter " + filterBy);
+                logger.info(res.toString());
+                continue;
+            }
+
+            switch (filter) {
+                case TRANSACTIONID:
+                    spec = TransactionRequestSpecs.in(TransactionRequest_.transactionId, ids);
+                    break;
+                case PAYERID:
+                    spec = TransactionRequestSpecs.in(TransactionRequest_.payerPartyId, ids);
+                    break;
+                case PAYEEID:
+                    spec = TransactionRequestSpecs.in(TransactionRequest_.payeePartyId, ids);
+                    break;
+                case WORKFLOWINSTANCEKEY:
+                    spec = TransactionRequestSpecs.in(TransactionRequest_.workflowInstanceKey, ids);
+                    break;
+                case STATE:
+                    spec = TransactionRequestSpecs.in(TransactionRequest_.state, parseStates(ids));
+                    break;
+                case ERRORDESCRIPTION:
+                    spec = TransactionRequestSpecs.filterByErrorDescription(parseErrorDescription(ids));
+                    break;
+            }
 
-        } else {
             PageRequest pager = new PageRequest(page, size, new Sort(Sort.Direction.valueOf(sortedOrder), "startedAt"));
             Page<TransactionRequest> result;
             if (spec == null) {
                 result = transactionRequestRepository.findAll(pager);
             } else {
+                for (int i = 1; i < specs.size(); i++) {
+                    spec = spec.and(specs.get(i));
+                }
                 result = transactionRequestRepository.findAll(spec, pager);
             }
-            data = result.getContent();
+            data.addAll(result.getContent());
```

Comment:
Note that result.getContent and Page are all using the domain object TransactionRequest. Can you double check once why ExternalId (mpesaReceiptId) column is not coming in CSV?

Reply:
I think that field is not populated in mysql.

Reply:
done